src/plugins/cjs-interop.ts

```diff
@@ -10,17 +15,81 @@ export const cjsInterop = (): Plugin => {
         this.format !== 'cjs' ||
         info.type !== 'chunk' ||
         !/\.(js|cjs)$/.test(info.path) ||
-        !info.entryPoint ||
-        info.exports?.length !== 1 ||
-        info.exports[0] !== 'default'
+        !info.entryPoint
       ) {
         return
       }
 
+      if (this.splitting) {
+        // there is exports metadata when cjs+splitting is set
+        if (info.exports?.length !== 1 || info.exports[0] !== 'default') return
+      } else {
+        const swc: typeof import('@swc/core') = localRequire('@swc/core')
+        const { Visitor }: typeof import('@swc/core/Visitor') =
+          localRequire('@swc/core/Visitor')
+        if (!swc || !Visitor) {
+          throw new PrettyError(
+            `@swc/core is required for cjsInterop when splitting is not enabled. Please install it with \`npm install @swc/core -D\``
+          )
+        }
+
+        let entrySource: string | undefined
+        try {
+          entrySource = await fs.readFile(info.entryPoint!, {
+            encoding: 'utf8',
+          })
+        } catch {}
+        if (!entrySource) return
+
+        const ast = await swc.parse(entrySource, {
+          syntax: 'typescript',
```

Comment:
check extname for emcascript

Reply:
just curious, why do we need this? isn't tsx a superset of ecmascript?

Reply:
In certain special cases, it is not.

Reply:
could you provide some examples so that I can refine the unit test cases?

Reply:
I don't quite remember now, and I can't provide an example at the moment. But just to be safe, we'd better make a distinction.

Reply:
Considering JSX and TSX, they are definitely not in a subset relationship because type assertion and JSX tags conflict.

```jsx
const a = <Type>b
```